contrib/runners/orquesta_runner/tests/unit/test_output_schema.py

```diff
@@ -0,0 +1,170 @@
+        wf_ex_db = wf_db_access.WorkflowExecution.get_by_id(wf_ex_db.id)
+        query_filters = {'workflow_execution': str(wf_ex_db.id), 'task_id': 'task3'}
+        tk3_ex_db = wf_db_access.TaskExecution.query(**query_filters)[0]
+        tk3_ac_ex_db = ex_db_access.ActionExecution.query(task_execution=str(tk3_ex_db.id))[0]
+        wf_svc.handle_action_execution_completion(tk3_ac_ex_db)
+        wf_ex_db = wf_db_access.WorkflowExecution.get_by_id(wf_ex_db.id)
+        lv_ac_db = lv_db_access.LiveAction.get_by_id(str(lv_ac_db.id))
+        ac_ex_db = ex_db_access.ActionExecution.get_by_id(str(ac_ex_db.id))
+
+        config = self.loader(os.path.join(BASE_DIR, '../../runner.yaml'))
+
+        output_schema._validate_runner(config[0]['output_schema'], lv_ac_db.result)
+
+    def test_fail_incorrect_output_schema(self):
+        wf_meta = base.get_wf_fixture_meta_data(TEST_PACK_PATH, 'sequential.yaml')
+        wf_input = {'who': 'Thanos'}
+        lv_ac_db = lv_db_models.LiveActionDB(action=wf_meta['name'], parameters=wf_input)
+        lv_ac_db, ac_ex_db = ac_svc.request(lv_ac_db)
+        lv_ac_db = lv_db_access.LiveAction.get_by_id(str(lv_ac_db.id))
+        wf_ex_dbs = wf_db_access.WorkflowExecution.query(action_execution=str(ac_ex_db.id))
+        wf_ex_db = wf_ex_dbs[0]
+        query_filters = {'workflow_execution': str(wf_ex_db.id), 'task_id': 'task1'}
+        tk1_ex_db = wf_db_access.TaskExecution.query(**query_filters)[0]
+        tk1_ac_ex_db = ex_db_access.ActionExecution.query(task_execution=str(tk1_ex_db.id))[0]
+        wf_svc.handle_action_execution_completion(tk1_ac_ex_db)
+        tk1_ex_db = wf_db_access.TaskExecution.get_by_id(tk1_ex_db.id)
+        wf_ex_db = wf_db_access.WorkflowExecution.get_by_id(wf_ex_db.id)
+        query_filters = {'workflow_execution': str(wf_ex_db.id), 'task_id': 'task2'}
+        tk2_ex_db = wf_db_access.TaskExecution.query(**query_filters)[0]
+        tk2_ac_ex_db = ex_db_access.ActionExecution.query(task_execution=str(tk2_ex_db.id))[0]
+        wf_svc.handle_action_execution_completion(tk2_ac_ex_db)
+        tk2_ex_db = wf_db_access.TaskExecution.get_by_id(tk2_ex_db.id)
+        wf_ex_db = wf_db_access.WorkflowExecution.get_by_id(wf_ex_db.id)
+        query_filters = {'workflow_execution': str(wf_ex_db.id), 'task_id': 'task3'}
+        tk3_ex_db = wf_db_access.TaskExecution.query(**query_filters)[0]
+        tk3_ac_ex_db = ex_db_access.ActionExecution.query(task_execution=str(tk3_ex_db.id))[0]
+        wf_svc.handle_action_execution_completion(tk3_ac_ex_db)
+        wf_ex_db = wf_db_access.WorkflowExecution.get_by_id(wf_ex_db.id)
+        lv_ac_db = lv_db_access.LiveAction.get_by_id(str(lv_ac_db.id))
+        ac_ex_db = ex_db_access.ActionExecution.get_by_id(str(ac_ex_db.id))
+
+        with self.assertRaises(jsonschema.ValidationError):
+            output_schema._validate_runner(
+                FAIL_SCHEMA,
+                lv_ac_db.result
+            )
```

Comment:
This unit test looks wrong. If the output should fail schema validation, the lv_ac_db should be in a failed state with appropriate error message in the lv_ac_db.result['errors'].

Reply:
Failing w/ errors is still adhears to the output schema. But I will look through and change it to a succeeding WF.

Reply:
I see what you're saying, but no the workflow can succeed from the Workflow engine's perspective but when it's handed over for DB write (where schema is checked) then it should raise the schema error and rewrite the result to the appropriate output. The workflow shows as failed. This test takes a valid output from orquesta's perspective but uses a different schema and confirms that it will raise when the output doesn't adhere to the schema. Probably better test would be to change the output of the WF result to a non-orquesta output instead of the schema itself.